### PR TITLE
abcMIDI: init at 2017.06.10

### PR DESCRIPTION
**Pull request by dotlambda**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -13381,6 +13381,8 @@ with pkgs;
     libcdio = libcdio082;
   };
 
+  abcmidi = callPackage ../applications/audio/abcmidi { };
```
Comment (rycee) [inline pkgs/top-level/all-packages.nix]: If I understand it correctly then abcmidi is mostly a collection of command line tools. If so then I would suggest to use `../tools/audio/abcmidi` instead.

Comment (dotlambda) [reply pkgs/top-level/all-packages.nix]: You're absolutely correct. Corrected that.
Meanwhile, I don't really get the ordering in `all-packages.nix`.

Comment (rycee) [reply pkgs/top-level/all-packages.nix]: It is … complicated. Basically, stick your thing in roughly the right place and don't worry about it.

**Comment by rycee** [comment]
Rebased to master in 608045aed54c3579923d54c533b2d57a09c361e1. Thanks for the contribution!

**Pull request closed without merging**